### PR TITLE
[test_nn] split lazy_modules from test_nn

### PR DESCRIPTION
**Pull request by kshitij12345**
Ref: #63085

NOTE: We don't need an accompanying XLA PR as these tests run only on CPU.

### CODE REVIEW

File: test/nn/test_lazy_modules.py [hunk 1]
```diff
@@ -0,0 +1,625 @@
...
+    def test_lazy_instancenorm3d_state(self):
+        self._check_lazy_instancenorm_state(nn.InstanceNorm3d, nn.LazyInstanceNorm3d)
+        self._check_lazy_instancenorm_state(nn.InstanceNorm3d, nn.LazyInstanceNorm3d)
+
+    @suppress_warnings
+    def test_materialize_dtype(self):
+        module = LazyModule()
+        module.register_parameter('test_param', UninitializedParameter())
+        module.test_param.materialize(10)
+        self.assertTrue(module.test_param.dtype == torch.get_default_dtype())
```
Comment (kshitij12345) [inline test/nn/test_lazy_modules.py:529]: It used to be 
`self.assertTrue(module.test_param.dtype == torch.float64)` as test_nn sets default dtype to `torch.float64`.

Comment (albanD) [reply test/nn/test_lazy_modules.py:530]: Thanks for pointing it out!
Fixing these tests to not use float64 by default is definitely something we want.

**Review by albanD — APPROVED** [review]
SGTM

**Pull request closed without merging**